`AttributeError` for `pytest v7.2.0`

[Failing CI](https://github.com/beeware/briefcase/actions/runs/3324020664/jobs/5495092973) for beeware/briefcase#943 is suggesting `pytest-tldr` needs updates to continue working with `pytest>=7.2.0`.

```
11s
Run tox -e py --installpkg dist/*.whl
  tox -e py --installpkg dist/*.whl
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    python_version: 3.9
    pythonLocation: /opt/hostedtoolcache/Python/3.[1](https://github.com/beeware/briefcase/actions/runs/3324020664/jobs/5495092973#step:6:1)0.8/x64
    PKG_CONFIG_PATH: /opt/hostedtoolcache/Python/3.10.8/x64/lib/pkgconfig
    Python_ROOT_DIR: /opt/hostedtoolcache/Python/3.10.8/x64
    Python[2](https://github.com/beeware/briefcase/actions/runs/3324020664/jobs/5495092973#step:6:2)_ROOT_DIR: /opt/hostedtoolcache/Python/[3](https://github.com/beeware/briefcase/actions/runs/3324020664/jobs/5495092973#step:6:3).10.8/x6[4](https://github.com/beeware/briefcase/actions/runs/3324020664/jobs/5495092973#step:6:4)
    Python3_ROOT_DIR: /opt/hostedtoolcache/Python/3.10.8/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.10.8/x64/lib
py create: /home/runner/work/briefcase/briefcase/.tox/py
py inst: /home/runner/work/briefcase/briefcase/.tox/.tmp/package/1/briefcase-0.3.12.dev[5](https://github.com/beeware/briefcase/actions/runs/3324020664/jobs/5495092973#step:6:5)1+gd4888321-py3-none-any.whl
py installed: arrow==1.2.3,attrs==22.1.0,binaryornot==0.4.4,briefcase @ file:///home/runner/work/briefcase/briefcase/.tox/.tmp/package/1/briefcase-0.3.12.dev51%2Bgd4888321-py3-none-any.whl,certifi==2022.9.24,chardet==5.0.0,charset-normalizer==2.1.1,click==8.1.3,commonmark==0.9.1,cookiecutter==2.1.1,coverage==[6](https://github.com/beeware/briefcase/actions/runs/3324020664/jobs/5495092973#step:6:6).5.0,exceptiongroup==1.0.0rc9,gitdb==4.0.9,GitPython==3.1.29,idna==3.4,iniconfig==1.1.1,Jinja2==3.1.2,jinja2-time==0.2.0,MarkupSafe==2.1.1,packaging==21.3,platformdirs==2.5.2,pluggy==1.0.0,psutil==5.9.3,Pygments==2.13.0,pyparsing==3.0.9,pytest==[7](https://github.com/beeware/briefcase/actions/runs/3324020664/jobs/5495092973#step:6:7).2.0,pytest-cov==4.0.0,pytest-tldr==0.2.4,python-dateutil==2.[8](https://github.com/beeware/briefcase/actions/runs/3324020664/jobs/5495092973#step:6:8).2,python-slugify==6.1.2,PyYAML==6.0,requests==2.28.1,rich==12.6.0,six==1.16.0,smmap==5.0.0,text-unidecode==1.3,tomli==2.0.1,tomli_w==1.0.0,urllib3==1.26.12
py run-test-pre: PYTHONHASHSEED='420284[9](https://github.com/beeware/briefcase/actions/runs/3324020664/jobs/5495092973#step:6:9)385'
py run-test: commands[0] | python -m coverage run -m pytest -vv
platform linux -- Python 3.[10](https://github.com/beeware/briefcase/actions/runs/3324020664/jobs/5495092973#step:6:10).8
pytest==7.2.0

INTERNALERROR> Traceback (most recent call last):

INTERNALERROR>   File "/home/runner/work/briefcase/briefcase/.tox/py/lib/python3.10/site-packages/_pytest/main.py", line 268, in wrap_session

INTERNALERROR>     config.hook.pytest_sessionstart(session=session)

INTERNALERROR>   File "/home/runner/work/briefcase/briefcase/.tox/py/lib/python3.10/site-packages/pluggy/_hooks.py", line 265, in __call__

INTERNALERROR>     return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)

INTERNALERROR>   File "/home/runner/work/briefcase/briefcase/.tox/py/lib/python3.10/site-packages/pluggy/_manager.py", line 80, in _hookexec

INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)

INTERNALERROR>   File "/home/runner/work/briefcase/briefcase/.tox/py/lib/python3.10/site-packages/pluggy/_callers.py", line 60, in _multicall

INTERNALERROR>     return outcome.get_result()

INTERNALERROR>   File "/home/runner/work/briefcase/briefcase/.tox/py/lib/python3.10/site-packages/pluggy/_result.py", line 60, in get_result

INTERNALERROR>     raise ex[1].with_traceback(ex[2])

INTERNALERROR>   File "/home/runner/work/briefcase/briefcase/.tox/py/lib/python3.10/site-packages/pluggy/_callers.py", line 39, in _multicall

INTERNALERROR>     res = hook_impl.function(*args)

INTERNALERROR>   File "/home/runner/work/briefcase/briefcase/.tox/py/lib/python3.10/site-packages/pytest_tldr.py", line [15](https://github.com/beeware/briefcase/actions/runs/3324020664/jobs/5495092973#step:6:16)5, in pytest_sessionstart

INTERNALERROR>     self.print("py=={}".format(py.__version__))

INTERNALERROR> AttributeError: module 'py' has no attribute '__version__'
ERROR: InvocationError for command /home/runner/work/briefcase/briefcase/.tox/py/bin/python -m coverage run -m pytest -vv (exited with code 3)
___________________________________ summary ____________________________________
ERROR:   py: commands failed
Error: Process completed with exit code 1.
```

This seems like the likely source (from 7.2.0 release notes):
> https://github.com/pytest-dev/pytest/issues/10396: pytest no longer depends on the py library. pytest provides a vendored copy of py.error and py.path modules but will use the py library if it is installed. If you need other py.* modules, continue to install the deprecated py library separately, otherwise it can usually be removed as a dependency.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -38,36 +38,24 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        tox-python-version: ['py37', 'py38', 'py39', 'py310', 'py311'] #, 'py312']
-        pytest-version: ['3', '4', '5', '6', '7']
-        include:
-          - tox-python-version: 'py37'
-            python-version: "3.7"
-          - tox-python-version: 'py38'
-            python-version: "3.8"
-          - tox-python-version: 'py39'
-            python-version: "3.9"
-          - tox-python-version: 'py310'
-            python-version: "3.10"
-          - tox-python-version: 'py311'
-            python-version: "3.11"
-          # - tox-python-version: 'py312'
-          #   python-version: "3.12.0-alpha - 3.12.0"
+        python-version: ["3.7", "3.8", "3.9", "3.10", "3.11"]
+        pytest-version: ["3", "4", "5", "6", "7"]
 
         exclude:
-          - tox-python-version: py310
-          - pytest-version: 3
-          - tox-python-version: py310
-          - pytest-version: 4
-          - tox-python-version: py310
-          - pytest-version: 5
+          # PyTest <=5 doesn't work on Python >= 3.10
+          - python-version: "3.10"
+            pytest-version: "3"
+          - python-version: "3.10"
+            pytest-version: "4"
+          - python-version: "3.10"
+            pytest-version: "5"
 
-          - tox-python-version: py311
-          - pytest-version: 3
-          - tox-python-version: py311
-          - pytest-version: 4
-          - tox-python-version: py311
-          - pytest-version: 5
+          - python-version: "3.11"
+            pytest-version: "3"
+          - python-version: "3.11"
+            pytest-version: "4"
+          - python-version: "3.11"
+            pytest-version: "5"
     steps:
     - uses: actions/checkout@v1
     - name: Set up Python ${{ matrix.python-version }}
@@ -81,4 +69,5 @@ jobs:
         python -m pip install tox
     - name: Test
       run: |
-        tox -e ${{ matrix.tox-python-version }}-pytest${{ matrix.pytest-version }}
+        export PY_VERSION=${{ matrix.python-version }}
+        tox -e py${PY_VERSION/./}-pytest${{ matrix.pytest-version }}

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -17,9 +17,11 @@ jobs:
         - 'towncrier-check'
         - 'package'
     steps:
-    - uses: actions/checkout@v1
+    - uses: actions/checkout@v3.1.0
+      with:
+        fetch-depth: 0
     - name: Set up Python
-      uses: actions/setup-python@v1
+      uses: actions/setup-python@v4.3.0
       with:
         python-version: '3.X'
     - name: Install dependencies
@@ -54,7 +56,7 @@ jobs:
     steps:
     - uses: actions/checkout@v1
     - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v1
+      uses: actions/setup-python@v4.3.0
       with:
         python-version: ${{ matrix.python-version }}
     - name: Install dependencies

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -38,7 +38,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        tox-python-version: ['py37', 'py38', 'py39', 'py310', 'py311', 'py312']
+        tox-python-version: ['py37', 'py38', 'py39', 'py310', 'py311'] #, 'py312']
         pytest-version: ['3', '4', '5', '6', '7']
         include:
           - tox-python-version: 'py37'
@@ -51,8 +51,8 @@ jobs:
             python-version: "3.10"
           - tox-python-version: 'py311'
             python-version: "3.11"
-          - tox-python-version: 'py312'
-            python-version: "3.12.0-alpha - 3.12.0"
+          # - tox-python-version: 'py312'
+          #   python-version: "3.12.0-alpha - 3.12.0"
     steps:
     - uses: actions/checkout@v1
     - name: Set up Python ${{ matrix.python-version }}

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -32,47 +32,27 @@ jobs:
       run: |
         tox -e ${{ matrix.task }}
 
-  smoke:
+  build:
     needs: beefore
     runs-on: ubuntu-latest
     strategy:
       max-parallel: 4
       matrix:
-        tox-python-version: ['py36']
-        pytest-version: ['3', '4', '5', '6']
-        include:
-          - tox-python-version: 'py36'
-            python-version: 3.6
-    steps:
-    - uses: actions/checkout@v1
-    - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v1
-      with:
-        python-version: ${{ matrix.python-version }}
-    - name: Install dependencies
-      run: |
-        python -m pip install --upgrade pip
-        python -m pip install --upgrade setuptools
-        python -m pip install tox
-    - name: Test
-      run: |
-        tox -e ${{ matrix.tox-python-version }}-pytest${{ matrix.pytest-version }}
-
-  build:
-    needs: smoke
-    runs-on: ubuntu-latest
-    strategy:
-      max-parallel: 4
-      matrix:
-        tox-python-version: ['py37', 'py38', 'py39']
-        pytest-version: ['3', '4', '5', '6']
+        tox-python-version: ['py37', 'py38', 'py39', 'py310', 'py311', 'py312']
+        pytest-version: ['3', '4', '5', '6', '7']
         include:
           - tox-python-version: 'py37'
-            python-version: 3.7
+            python-version: "3.7"
           - tox-python-version: 'py38'
-            python-version: 3.8
+            python-version: "3.8"
           - tox-python-version: 'py39'
-            python-version: 3.9
+            python-version: "3.9"
+          - tox-python-version: 'py310'
+            python-version: "3.10"
+          - tox-python-version: 'py311'
+            python-version: "3.11"
+          - tox-python-version: 'py312'
+            python-version: "3.12.0-alpha - 3.12.0"
     steps:
     - uses: actions/checkout@v1
     - name: Set up Python ${{ matrix.python-version }}

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -53,6 +53,21 @@ jobs:
             python-version: "3.11"
           # - tox-python-version: 'py312'
           #   python-version: "3.12.0-alpha - 3.12.0"
+
+        exclude:
+          - tox-python-version: py310
+          - pytest-version: 3
+          - tox-python-version: py310
+          - pytest-version: 4
+          - tox-python-version: py310
+          - pytest-version: 5
+
+          - tox-python-version: py311
+          - pytest-version: 3
+          - tox-python-version: py311
+          - pytest-version: 4
+          - tox-python-version: py311
+          - pytest-version: 5
     steps:
     - uses: actions/checkout@v1
     - name: Set up Python ${{ matrix.python-version }}

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -3,14 +3,13 @@ on:
   pull_request:
   push:
     branches:
-      - master
+      - main
 
 jobs:
   beefore:
     name: Pre-test checks
     runs-on: ubuntu-latest
     strategy:
-      max-parallel: 4
       matrix:
         task:
         - 'flake8'
@@ -36,7 +35,6 @@ jobs:
     needs: beefore
     runs-on: ubuntu-latest
     strategy:
-      max-parallel: 4
       matrix:
         tox-python-version: ['py37', 'py38', 'py39', 'py310', 'py311', 'py312']
         pytest-version: ['3', '4', '5', '6', '7']

--- a/changes/36.bugfix.rst
+++ b/changes/36.bugfix.rst
@@ -1,0 +1,1 @@
+An incompatibility with Pytest 7.2 was corrected.

--- a/changes/template.rst
+++ b/changes/template.rst
@@ -1,7 +1,9 @@
 {% for section, _ in sections.items() %}
 {% set underline = underlines[0] %}{% if section %}{{section}}
 {{ underline * section|length }}{% set underline = underlines[1] %}
+
 {% endif %}
+
 {% if sections[section] %}
 {% for category, val in definitions.items() if category in sections[section]%}
 {{ definitions[category]['name'] }}
@@ -10,16 +12,18 @@
 {% if definitions[category]['showcontent'] %}
 {% for text, values in sections[section][category].items() %}
 * {{ text }} ({{ values|join(', ') }})
-
 {% endfor %}
+
 {% else %}
 * {{ sections[section][category]['']|join(', ') }}
+
 {% endif %}
 {% if sections[section][category]|length == 0 %}
 No significant changes.
 
 {% else %}
 {% endif %}
+
 {% endfor %}
 {% else %}
 No significant changes.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -10,6 +10,4 @@ directory = "changes"
 package = "pytest_tldr"
 filename = "CHANGELOG.rst"
 title_format = "{version} ({project_date})"
-issue_format = "`#{issue} <https://github.com/freakboy3742/pytest-tldr/issues/{issue}>`_"
 template = "changes/template.rst"
-underlines = ["-", "^", "\""]

--- a/pytest_tldr.py
+++ b/pytest_tldr.py
@@ -152,7 +152,12 @@ class TLDRReporter:
                 msg += "[pypy-{}-{}]".format(verinfo, sys.pypy_version_info[3])
             self.print(msg)
             self.print("pytest=={}".format(pytest.__version__))
-            self.print("py=={}".format(py.__version__))
+            try:
+                # Pytest 7.2 vendored `py`; if it's vendored, the version
+                # won't exist, but we also don't care that it doesn't exist.
+                self.print("py=={}".format(py.__version__))
+            except AttributeError:
+                pass
             self.print("pluggy=={}".format(pluggy.__version__))
 
             headers = self.config.hook.pytest_report_header(

--- a/setup.cfg
+++ b/setup.cfg
@@ -20,6 +20,8 @@ classifiers =
     Programming Language :: Python :: 3.7
     Programming Language :: Python :: 3.8
     Programming Language :: Python :: 3.9
+    Programming Language :: Python :: 3.10
+    Programming Language :: Python :: 3.11
     Programming Language :: Python :: 3 :: Only
     Programming Language :: Python :: Implementation :: CPython
     Programming Language :: Python :: Implementation :: PyPy

--- a/tox.ini
+++ b/tox.ini
@@ -4,7 +4,7 @@
 # and then run "tox" from this directory.
 
 [tox]
-envlist = flake8,check-manifest,towncrier-check,package,py{36,37,38,39}-pytest{3,4,5,6}
+envlist = flake8,check-manifest,towncrier-check,package,py{37,38,39,310,311,312}-pytest{3,4,5,6,7}
 skip_missing_interpreters = true
 
 [testenv]
@@ -12,7 +12,8 @@ deps =
     pytest3: pytest < 4
     pytest4: pytest < 5
     pytest5: pytest < 6
-    pytest6: pytest >= 6.0.0
+    pytest6: pytest < 7
+    pytest7: pytest >= 6.0.0
     pytest-cov
     pytest-tldr
 commands =

--- a/tox.ini
+++ b/tox.ini
@@ -35,14 +35,14 @@ commands =
 [testenv:towncrier-check]
 skip_install = True
 deps =
-    towncrier >= 18.5.0
+    towncrier ~= 22.8
 commands =
    python -m towncrier.check
 
 [testenv:towncrier]
 skip_install = True
 deps =
-    towncrier >= 18.5.0
+    towncrier ~= 22.8
 commands =
     towncrier {posargs}
 

--- a/tox.ini
+++ b/tox.ini
@@ -37,7 +37,7 @@ skip_install = True
 deps =
     towncrier ~= 22.8
 commands =
-   python -m towncrier.check
+   python -m towncrier.check --compare-with origin/main
 
 [testenv:towncrier]
 skip_install = True

